scrypto-derive/src/blueprint.rs

```diff
@@ -499,11 +501,55 @@ fn generate_stubs(
             }
         }
 
-        impl #value_ident {
+        impl #component_ident {
             #(#functions)*
 
             #(#methods)*
         }
+
+        #[allow(non_camel_case_types)]
+        pub struct #component_ref_ident {
+            pub component: ::scrypto::component::GlobalComponentRef,
+        }
+
+        impl From<ComponentAddress> for #component_ref_ident {
+            fn from(component: ComponentAddress) -> Self {
+                Self {
+                    component: ::scrypto::component::GlobalComponentRef(component)
+                }
+            }
+        }
+
+        impl ::scrypto::component::GlobalComponent for #component_ref_ident {
+            fn package_address(&self) -> ::scrypto::model::PackageAddress {
+                self.component.package_address()
+            }
+            fn blueprint_name(&self) -> String {
+                self.component.blueprint_name()
+            }
+            fn metadata<K: AsRef<str>, V: AsRef<str>>(&mut self, name: K, value: V) -> &mut Self {
+                self.component.metadata(name, value);
+                self
+            }
+            fn add_access_check(&mut self, access_rules: ::scrypto::model::AccessRules) -> &mut Self {
+                self.component.add_access_check(access_rules);
+                self
+            }
+            fn set_royalty_config(&mut self, royalty_config: ::scrypto::model::RoyaltyConfig) -> &mut Self {
+                self.component.set_royalty_config(royalty_config);
+                self
+            }
+            fn claim_royalty(&self) -> Bucket {
+                self.component.claim_royalty()
+            }
+            fn access_rules_chain(&self) -> Vec<ComponentAccessRules> {
+                self.component.access_rules_chain()
+            }
+        }
+
+        impl #component_ref_ident {
+            #(#methods)*
```

Comment:
Is `#(#functions)*` intentionally left out here?

Reply:
Yup. Didn't want to add all stubs to the global version.